### ISSUE TITLE
Handle unwanted writes in debugger

### ISSUE THREAD

**Issue by thrust26** [stella-emu/stella#396]
Reading from e.g. SuperChip RAM using a write address, this causes an unwanted write to this address. This coding error can currently no be debugged efficiently (e.g. via trap).

The debugger should break on such unwanted writes (enabled via Developer option) and print a message.

**Comment by sa666666** [comment]
Seems easy enough.  Do we want to expose the info through `M6502`/`System` so that `trapwrite` will catch it too, or is it sufficient to just pop up the debugger with a message?

**Comment by thrust26** [comment]
I think the latter will do. Combined with a Developer setting (it must not break for normal players).

**Comment by thrust26** [comment]
BTW: We have the `_rwport` command, which returns an address in case such an unwanted write happened since the last check.

**Comment by sa666666** [comment]
Yep, that's why I mentioned it would be easy to do this.  We already detect the condition, and have a pseudo-op to check against.  Opening the debugger with a message is a minor addition.

**Comment by sa666666** [comment]
I am going to attempt this.  @thrust26, are you doing it instead (I noticed you're assigned to it)?  Either way, I don't want us both to waste time working on the same thing.

**Comment by thrust26** [comment]
Already started working on it. Currently debugging...

**Comment by sa666666** [comment]
OK.

**Comment by thrust26** [comment]
@sa666666 Your refactoring works well, but I wonder why you omitted the extra check like:
```
// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
int CartDebug::readFromWritePort()
{
  uInt16 addr = myRWPortAddress;
  myRWPortAddress = 0;

  // A read from the write port occurs when the read is actually in the write
  // port address space AND the last access was actually a read (the latter
  // differentiates between reads that are normally part of a write cycle vs.
  // ones that are illegal)
  if(mySystem.m6502().lastReadAddress() &&
      (mySystem.getPageAccessType(addr) & System::PA_WRITE) == System::PA_WRITE)
    return addr;
  else
    return 0;
}
```
Are they not relevant?

**Comment by thrust26** [comment]
This is still not correct. It triggers when writing to a legal address. Still the CPU peeks seem to be the problem.

**Comment by sa666666** [comment]
Oops, missed that part.  We still need to get this fixed though, especially if _rwport was broken all along.
